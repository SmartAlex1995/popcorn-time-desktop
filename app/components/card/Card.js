/**
 * Card in the CardList component
 * @flow
 */
import React from 'react';
import { Col, Row } from 'reactstrap';
import { Link } from 'react-router-dom';
import Rating from './Rating';

type Props = {
  title: string,
  starColor?: string,
  image: string,
  id: string,
  rating: number | 'n/a',
  type: string
};

export default function Card(props: Props) {
  const { type, image, id, rating, title, starColor } = props;

  const placeholder =
    process.env.NODE_ENV === 'production'
      ? './images/posterholder.png'
      : './app/images/posterholder.png';

  const backgroundImageStyle = {
    backgroundImage: `url(${image !== 'n/a' ? image : placeholder})`
  };

  return (
    <div className="Card">
      <Link replace to={`/item/${type}/${id}`}>
        <div className="Card--overlay-container" style={backgroundImageStyle}>
          <div className="Card--overlay" />
        </div>
      </Link>
      <div className="Card--descrption">
        <Row>
          <Col sm="12">
            <Link className="Card--title" replace to={`/item/${type}/${id}`}>
              {title}
            </Link>
          </Col>
        </Row>
        <Row>
          <Col sm="12">
            {typeof rating === 'number' && (
              <Rating starColor={starColor} rating={rating} />
            )}
          </Col>
        </Row>
      </div>
    </div>
  );
}

Card.defaultProps = {
  starColor: '#848484'
};
