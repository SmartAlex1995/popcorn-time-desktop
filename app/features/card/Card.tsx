/**
 * Card in the CardsGrid component
 */
import React from "react";
import { Col, Row } from "reactstrap";
import { Link } from "react-router-dom";
import Rating from "./Rating";
import { ItemKind } from "../../api/metadata/MetadataProviderInterface";

type Props = {
  title: string;
  starColor?: string;
  image: string;
  id: string;
  rating: number;
  type: ItemKind;
};

export default function Card(props: Props) {
  const { type, image, id, rating, title, starColor } = props;

  const placeholder =
    process.env.NODE_ENV === "production"
      ? "./images/posterholder.png"
      : "./app/images/posterholder.png";

  const backgroundImageStyle = {
    backgroundImage: `url(${image !== "n/a" ? image : placeholder})`,
  };

  return (
    <div className="Card">
      <Link to={`/${type}/${id}`}>
        <div className="Card--overlay-container" style={backgroundImageStyle}>
          <div className="Card--overlay" />
        </div>
      </Link>
      <div className="Card--descrption">
        <Row>
          <Col sm="12">
            <Link className="Card--title" to={`/${type}/${id}`}>
              {title}
            </Link>
            {typeof rating === "number" && (
              <Rating starColor={starColor} rating={rating} />
            )}
          </Col>
        </Row>
      </div>
    </div>
  );
}

Card.defaultProps = {
  starColor: "#848484",
};
