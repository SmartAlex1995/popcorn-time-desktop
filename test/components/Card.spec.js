import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Card from '../../app/components/card/Card.jsx';

function setup(propsOverride) {
  const props = Object.assign(
    {},
    {
      image: 'test_image_url',
      title: 'Test movie',
      id: '248245',
      genres: ['action', 'comedy'],
      rating: 4.5,
      kind: 'shows',
      baseUrl: '/item/shows'
    },
    propsOverride
  );

  const component = shallow(<Card {...props} />);

  return {
    component,
    title: component.find('.Card--title')
  };
}

describe('Card Component', () => {
  it('should have movie title', () => {
    const { title } = setup();
    expect(toJson(title)).toMatchSnapshot();
  });
});
