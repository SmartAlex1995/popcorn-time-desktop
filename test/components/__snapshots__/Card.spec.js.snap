// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should have movie title 1`] = `
<Link
  className="Card--title"
  to="/undefined/248245"
>
  Test movie
</Link>
`;
